crates/resolver/src/caching_client.rs

```diff
@@ -376,39 +381,59 @@ where
     }
 
     #[allow(clippy::unnecessary_wraps)]
-    fn cname(&self, lookup: Lookup, query: Query, cname_ttl: u32) -> Result<Lookup, ProtoError> {
-        // this duplicates the cache entry under the original query
-        Ok(self.lru.duplicate(query, lookup, cname_ttl, Instant::now()))
+    fn cname(&self, lookup: Lookup, query: Query) -> Result<Lookup, ProtoError> {
+        let mut message = Message::response(0, OpCode::Query);
```

Comment:
How good are we at appropriately setting the response ID especially when returning cached responses? Could/should we improve the API to force this?